### ISSUE TITLE
[Bug]:  Presets, opacity, blur not working

### ISSUE THREAD

**Issue by michaldybczak** [luisbocanegra/plasma-panel-colorizer#108]
- [X ] I have searched if this issue hasn't been reported yet.

**Describe the bug**
On the Presets tab, there is nothing to choose. When down arrow is clicked to unwrap the options, some weird, transparent shadow shows instead an extension with options.

![Screenshot_2024 10 27_05 31 36](https://github.com/user-attachments/assets/685a4a03-314f-49e3-a6d6-153bc5ea2572)

In Appearance tab, no matter what Opacity level I select (from 0 to 1), nothing is changing. Panel is still solid. Also, when I enable blur, also see no results.

The lack of options or not working elements may have the same underlying issue, since this is suspicious that so many things are missing at the same time, or be different bugs.

**To Reproduce**
1. Install applet
2. Add it on the panel.
3. Enable applet.
4. In settings, presets or their options are not available and other options like Opacity or Blur have no effect. When Background is disabled, also no results, the panel is still solid.

**Expected behavior**
I expect some options to show up under presets options and other mentioned above options should work.

**Screenshots**

**Project version**
You can find this in the *Widget settings* > *About* page [e.g. 0.5.2]

**Desktop (please complete the following information):**
Operating System: Manjaro Linux 
KDE Plasma Version: 6.2.2
KDE Frameworks Version: 6.7.0
Qt Version: 6.8.0
Kernel Version: 6.6.58-1-MANJARO (64-bit)
Graphics Platform: Wayland
Processors: 16 × AMD Ryzen 7 7840HS w/ Radeon 780M Graphics
Memory: 30.6 GiB of RAM
Graphics Processor: AMD Radeon 780M
Manufacturer: TUXEDO
Product Name: TUXEDO Sirius 16 Gen1

**Additional context**
My current global theme is Breeze Dark, so it was all very vanilla when I added the upper panel and colorizer applet.

I'm not sure what am I doing wrong. From issues reports here, people have no such complaints, so there hast to be some problem on my side, but I don't know which one. I just installed and added the applet and theming is pretty much vanilla, no Kvantum engine. Applet was installed/compiled from AUR.

**Comment by michaldybczak** [comment]
It turned out, the blur settings were set to strong, so any modifications were basically not visible, Actually, blur settings seems to do very little, between light or strong settings, so to see anything on the panel, I had to lower them to the maximum.

So we can rule out any problems with opacity. It does changes, but the differences are very mild. I may try some alternative blur effects from kwin-effects-forceblur.

However, I would love to see panel becoming solid with maximized windows. Now, it is only possible if opacity is set to 1. Anything below will stay partially transparent. I would expect that if preset will work, this issue will be fixed, unless I may not understand how presets work. However, I can't learn how they work since they don't work at all.

So the main issue is with presets.

**Comment by luisbocanegra** [comment]
> So why they do not show on Presets Autoloading tab?

Hi, Which version of the widget are you using? version [v1.0.0](https://github.com/luisbocanegra/plasma-panel-colorizer/releases/tag/v1.0.0) had a [bug listing presets on some distros](https://github.com/luisbocanegra/plasma-panel-colorizer/issues/99) but was fixed in [v1.0.1](https://github.com/luisbocanegra/plasma-panel-colorizer/releases/tag/v1.0.1)

**Comment by michaldybczak** [comment]
I don't have  ~/.local/share/plasma/plasmoids/luisbocanegra.panel.colorizer/contents/ui/configPresets.qml

Basically, I have ~/.local/share/plasma/plasmoids/ but there is no luisbocanegra.panel.colorize there.

When checked "cournalctl -f" it says that there is no such command.

Restart didn't help (it is a new install).

**Comment by luisbocanegra** [comment]
My bad it would be `/usr/share/plasma/plasmoids/luisbocanegra.panel.colorizer/contents/ui/configPresets.qml` I can reproduce after uninstalling the local version, it's a permissions problem:

```
qml: '/usr/share/plasma/plasmoids/luisbocanegra.panel.colorizer/contents/ui/tools/list_presets.sh' '/usr/share/plasma/plasmoids/luisbocanegra.panel.colorizer/contents/ui/presets/';'/usr/share/plasma/plasmoids/luisbocanegra.panel.colorizer/contents/ui/tools/list_presets.sh' '/home/luis/.config/panel-colorizer/presets/' 126 0  /bin/sh: line 1: /usr/share/plasma/plasmoids/luisbocanegra.panel.colorizer/contents/ui/tools/list_presets.sh: Permission denied
                                               /bin/sh: line 1: /usr/share/plasma/plasmoids/luisbocanegra.panel.colorizer/contents/ui/tools/list_presets.sh: Permission denied
```

**Comment by michaldybczak** [comment]
So to fix it manually on my side is just copy the file and locations to my user space?

I'll give it a try.

**Comment by luisbocanegra** [comment]
> So to fix it manually on my side is just copy the file and locations to my user space?
> 
> I'll give it a try.

No need to do that, just give executable permission to the file with `sudo chmod +x /usr/share/plasma/plasmoids/luisbocanegra.panel.colorizer/contents/ui/tools/list_presets.sh`

**Comment by michaldybczak** [comment]
That fixed it!

Now I see some screenshots of already existing presets and everything is colorful and full of choices! 

Now I just need to experiment with those ;). Thanks.

So basically, the issue is that the PKAGBUILD doesn't apply the proper permissions during install.

**Comment by luisbocanegra** [comment]
> So basically, the issue is that the PKAGBUILD doesn't apply the proper permissions during install.

Yes, will update the package later today.

**Comment by luisbocanegra** [comment]
> I have now transparent panel when not touching or maximized and solid when touching or maximized. It works like I imagined it would be! Awesome! Thank you!

Enjoy! Btw I agree about your point about screenshots, so will make them a prompt or separate button https://github.com/luisbocanegra/plasma-panel-colorizer/issues/109

**Comment by michaldybczak** [comment]
Yeah. When I first saved a preset, I didn't expect it to trigger a screenshot and didn't understand what is happening and why. It was especially confusing for me, because there were no default presets on Presets tab. I guess this element could be improved, but once the default presets are visible, the screenshot option makes sense.

I guess the problem is fixed, so I'm closing the issue.

Thank you very much. I didn't expect getting the solution so sun, it's Sunday evening after all.  I'm also happy with the applet, because now I have most of the things from latte.

There are some minor lacks, but in overall, buttons, title, global menu, colorizer and manual kwinrc config to getting rid of titlebar when windows are maximized can provide most of the latte features. Finally, I can restore my favorite setup after months of having, plain, default breeze layout. I missed global menus so much. Hamburger menus in titlebars were incredibly annoying to work with.
